libs/optimizer/src/lib/fixtures/optimizer.fixtures.ts

```diff
@@ -0,0 +1,457 @@
+import { OptimizationRequest, OptimizationResult, CircuitType } from '../optimizer';
+import { computeDestinationPoint, getGreatCircleBearing, getPreciseDistance } from 'geolib';
+import { createSegments } from '../utils/createSegments';
+import { concatTracks } from '../utils/concatTracks';
+import { ScoringRules } from '../scoringRules';
+
+/**
+ * Functions to generate test fixtures for the optimizer tests
+ */
+
+/**
+ * Test fixture for the optimizer tests
+ * givenRequest: the OptimizationRequest to test the optimizer
+ * expectedResult: the expected OptimizationResult that should be returned by the optimizer
+ */
+export type OptimizerFixture = {
+  givenRequest: OptimizationRequest;
+  givenRules: ScoringRules;
+  expectedResult: Omit<OptimizationResult, 'closingRadius'|'solutionIndices'>;
+};
+
+/**
+ * returns an empty track and its expected score
+ */
+export function createEmptyTrackFixture(): OptimizerFixture {
+  return {
+    givenRequest: {
+      track: { points: [], startTimeSec: 0 },
+    },
+    givenRules: ScoringRules.FederationFrancaiseVolLibre,
+    expectedResult: {
+      score: 0,
+      lengthKm: 0,
+      multiplier: 0,
+      circuit: undefined,
+      optimal: true,
+    },
+  };
+}
+
+export type LatLon = {
+  lat: number;
+  lon: number;
+};
+
+const START_TIME_SEC = Math.round(new Date().getTime() / 1000);
+/**
+ * @returns a fixture for a free distance track and it's expected score
+ * @param from LatLon of the starting point of the free distance
+ * @param to LatLon of the ending point of the free distance
+ * @param nbSegments number of segments to create between the two points
+ * @param givenRules the ScoringRules for computing the score
+ */
+export function createFreeDistanceFixture(
+  from: LatLon,
+  to: LatLon,
+  nbSegments: number,
+  givenRules: ScoringRules,
+): OptimizerFixture {
+  const multiplier = getFreeDistanceMultiplier(givenRules);
+  const distance = getPreciseDistance(from, to) / 1000;
+  return {
+    givenRequest: {
```

Comment:
nit: I think request and rules are good enough (no nee for "given") - up to you

Reply:
Ok. Removed `given` prefix